pyproject.toml

```diff
@@ -0,0 +1,35 @@
+[tool.codespell]
+ignore-words = "ignore_words.txt"
+uri-ignore-words-list = "bu"
+builtin = "clear,rare,informal,names,usage"
+skip = "doc/references.bib"
+
+[tool.ruff]
+select = ["E", "F", "W", "D"]
+exclude = ["__init__.py", "constants.py", "fixes.py", "resources.py"]
+ignore = ["D100", "D104", "D107", "D413"]
```

Comment:
```suggestion
ignore = [
    "D100", # Missing docstring in public module
    "D104", # Missing docstring in public package
    "D107", # Missing docstring in __init__
    "D413", # Missing blank line after last section
]
```

Reply:
if we're excluding `__init__.py` why do we need D107?  Is the `exclude` only excluding the main `__init__.py` for the whole package, and the `ignore` still applys to submodule `__init__.py`s?

Reply:
I'll test locally after committing your suggestions